app/uk/gov/hmrc/gform/gform/EmailAuthController.scala

```diff
@@ -329,6 +332,12 @@ class EmailAuthController(
             )
           )
           .map(_ => emailAndCode)
+
+      case (_: EmailAuthConfig, false) =>
```

Comment:
emailAuthEnabled should be renamed generateTestCode. When generateTestCode=true, we should response with ABCD, so this condition needs to be reversed